x-pack/plugins/endpoint/server/routes/resolver/utils/pagination.ts

```diff
@@ -15,6 +15,13 @@ export interface PaginationParams {
   eventID?: string;
 }
 
+export interface PaginatedResults {
+  totals: Record<string, number>;
+  results: ResolverEvent[];
+  // content holder for any other extra aggregation counts
+  extras?: Record<string, Record<string, number>>;
```

Comment:
nit: Maybe rename this to aggregations?

Reply:
Really, this is just used as a generic catch all, right now it's solely used for aggregations in the `stats` query, but anything that implements its own `afterSearch` hook can dump things here.